fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/Select.java

```diff
@@ -145,6 +161,109 @@ public String toString() {
             result.append(SPACE).append(this.orderByClause.toString());
         }
 
+        if (this.paginationClause != null) {
+            result.append(SPACE).append(this.paginationClause.toString());
+        }
+
         return result.toString();
     }
+
+    /**
+     * A string representation of the query with the bind variables substituted
+     * in place which is handy for debugging - but not to be used for actual
+     * execution.
+     * @return the query string starting XELECT instead of SELECT
+     */
+    public String toDebugString() {
```

Comment:
very interesting... X... keeps it from running?

Reply:
That was the idea, because it has the bind variables embedded as literals to make it easier for cut-and-paste debugging. I toyed with using XELECT so you didn't accidentally run a statement which might not be secure...but in the end I wanted to get the string with the bind variable markers anyway, so got rid of XELECT. Updated the comment.